libarchive/archive_read_open_file.c

```diff
@@ -191,6 +191,8 @@ FILE_seek(struct archive *a, void *client_data, int64_t request, int whence)
 		    (((int64_t)1 << (skip_bits - 1)) - 1) * 2 + 1;
 		if (request > max_skip)
 			skip = max_skip;
+		else if (request < ~max_skip)
```

Comment:
Huh?  I don't understand this part.

Reply:
Okay.  I thought about it and think I understand.  A comment is probably needed here.

Reply:
It's introduced for the possibility to seek backwards, i.e. if request is smaller than 0. Inverting the max value here should give us the minimum value of the data type.

If that's actually not possible, I can remove it.

Reply:
Adjusted. I use `min_seek` and declare it next to `max_seek` (instead of `max_skip` in the seek functions). Definitely improves readability. I have added a const to these variables so even badly optimizing compilers have a chance to do it right.